include/h2o.h

```diff
@@ -1322,6 +1332,10 @@ h2o_hostconf_t *h2o_req_setup(h2o_req_t *req);
  */
 void h2o_req_bind_conf(h2o_req_t *req, h2o_hostconf_t *hostconf, h2o_pathconf_t *pathconf);
 
+h2o_subreq_t *h2o_subrequest_create(h2o_req_t *parent, size_t sz);
```

Comment:
Could we have `h2o_subrequest_init` and `h2o_subrequest_dispose` instead? The reason I ask this is because the approach generally gives the user more freedom in how the object is allocated.

We would be constrained to "create" and "destroy" in some cases (e.g., when the size of the object changes depending on _how_ it is initialized), but we do not have such restriction in this case.